app/models/miq_server/worker_management/process.rb

```diff
@@ -38,18 +38,14 @@ def workers_quiesced?
   end
 
   def check_not_responding
-    worker_deleted = false
-    miq_workers.each do |w|
-      next unless monitor_reason_not_responding?(w)
-      next unless worker_get_monitor_status(w.pid) == :waiting_for_stop
+    workers_to_check = miq_workers.select do |w|
+      monitor_reason_not_responding?(w) &&
+        worker_get_monitor_status(w.pid) == :waiting_for_stop
+    end
```

Comment:
in contrast, this is only working on the sub select of records (calls a `next` otherwise). So converting over to a `select(&:)` makes sense - and I find it easier to read and reason.